CPS-????/README.md

```diff
@@ -0,0 +1,81 @@
+---
+CPS: ?
+Title: Spending Script Redundant Executions
+Status: Open
+Category: Plutus
+Authors: 
+  - fallen-icarus <modern.daidalos@gmail.com>
+Proposed Solutions: []
+Discussions:
+  - https://github.com/cardano-foundation/CIPs/issues/417
+Created: 2022-12-20
+---
+# CPS-XXXX: Spending Script Redundant Executions
+
+## Abstract
+Plutus spendings scripts are currently executed once for every UTxO being consumed from the script's address. However, there are use cases where the validity of the transaction depends on the transaction as a whole and not on any individual UTxO. The eUTxO model allows a spending script to see the entire transaction context which means it is already possible to create scripts that validate based on the transaction as a whole. As of now, spending scripts that do this are forced to be executed once per UTxO even if these extra executions are completely redundant. Not only does this waste computational resources, but it can also result in the transaction fee growing quickly based on the number of UTxOs being consumed from the relevant script address. There is a technique that can be used to minimize this limitation but it has its own drawbacks. 
```

Comment:
@fallen-icarus what technique can be used already to minimise this drawback?

Reply:
@matiwinnetou  The [Current Alternative](https://github.com/fallen-icarus/CIPs/tree/full-eUTxO-potential/CPS-%3F%3F%3F%3F#the-current-alternative) lays it out. In a nut shell, since you *can* make a plutus staking script execute only once for a given transaction, you off-load the heavy computation to that script. Then the spending plutus script just checks if the staking script was executed in the tx.

The idea is similar to what is being discussed in #309.